### ISSUE TITLE
`ktlint_code_style = android_studio` is broken

### ISSUE THREAD

**Issue by pdb-stripe** [pinterest/ktlint#2279]
## Expected Behavior
As documented at https://pinterest.github.io/ktlint/1.0.0/rules/code-styles/, using an `.editorconfig` containing this code should _work_ (produce compatibe checks).

```
[*.{kt,kts}]
ktlint_code_style = android_studio
```

## Observed Behavior
I upgraded from version 0.50.0 to 1.0.0 and used this in my .editorconfig. The resulting behavior was that I get false negatives. Ktlint produces an error about invalid ordering of imports, even though they are already lexographically organized without spaces. 

## Steps to Reproduce
Add the code snippet above to your editorconfig file, then execute ktlint. Note that I am using the [ktlint-gradle plugin](https://github.com/JLLeitschuh/ktlint-gradle) (v11.6.0), so execution includes the following:

`./gradlew --stop ; ./gradlew ktlintCheck`

## Your Environment

MacOS

## Additional Notes:

This works fine:

```
[*.{kt,kts}]
ktlint_code_style = intellij_idea
```

**Comment by paul-dingemans** [comment]
Please try to reproduce your problem with [ktlint cli](https://pinterest.github.io/ktlint/1.0.0/install/cli/) so that you can verify whether the problem is caused by ktlint or by the ktlint-gradle plugin.

I am sure that ktlint cli *does* recognize the `android_studio` code style.

**Comment by bjorn-jurgenn** [comment]
Same problem when using ktlint cli.

**Comment by paul-dingemans** [comment]
Please add code sample and editorconfig

**Comment by bjorn-jurgenn** [comment]
There is code sample

```
import androidx.compose.runtime.Composable

@Composable
fun SomeClass(
    parameter: String,
) {
    Text(text = "Hello $parameter", modifier = Modifier.padding(24.dp))
}

fun someMethod(
    parameter: String,
    parameter2: String,
) {
    println("someMethod")
}
```

And .editorconfig
```
[*.{kt,kts}]
ktlint_code_style = android_studio
```

Producing
```
/sample/SomeClass.kt:6:5: Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)
/sample/SomeClass.kt:7:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
/sample/SomeClass.kt:7:23: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
/sample/SomeClass.kt:13:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
/sample/SomeClass.kt:14:5: Single whitespace expected before parameter (standard:function-signature)
/sample/SomeClass.kt:14:24: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
```

**Comment by paul-dingemans** [comment]
> There is code sample
> 
> ```
> import androidx.compose.foundation.layout.padding
> import androidx.compose.material.Text
> import androidx.compose.runtime.Composable
> import androidx.compose.ui.Modifier
> import androidx.compose.ui.unit.dp
> 
> @Composable
> fun SampleComposable(
>     parameter: String,
>     modifier: Modifier = Modifier,
> ) {
>     Text(text = "Hello $parameter", modifier = Modifier.padding(24.dp))
> }
> 
> fun someMethod(
>     parameter: String,
> ) {
>     println("someMethod $parameter")
> }
> ```
> 
> And .editorconfig
> 
> ```
> [*.{kt,kts}]
> ktlint_code_style = android_studio
> ```
> 
> Producing
> 
> ```
> /sample/Sample.kt:10:5: Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)
> /sample/Sample.kt:11:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
> /sample/Sample.kt:12:5: Single whitespace expected before parameter (standard:function-signature)
> /sample/Sample.kt:12:35: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
> /sample/Sample.kt:18:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
> /sample/Sample.kt:18:23: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
> ```

How can you be sure that your issue is identical to OP? For your problem, see https://github.com/pinterest/ktlint/issues/2259#issuecomment-1732333419 which will be fixed in `1.0.1` release.

**Comment by paul-dingemans** [comment]
@pdb-stripe I will close this issue as it does not have enough details in title and body. I don't want this issue to be used as container for all problems with `1.0.0`. If your problem is not resolve, please add details and reopen the issue.

**Comment by bjorn-jurgenn** [comment]
> > There is code sample
> > ```
> > import androidx.compose.foundation.layout.padding
> > import androidx.compose.material.Text
> > import androidx.compose.runtime.Composable
> > import androidx.compose.ui.Modifier
> > import androidx.compose.ui.unit.dp
> > 
> > @Composable
> > fun SampleComposable(
> >     parameter: String,
> >     modifier: Modifier = Modifier,
> > ) {
> >     Text(text = "Hello $parameter", modifier = Modifier.padding(24.dp))
> > }
> > 
> > fun someMethod(
> >     parameter: String,
> > ) {
> >     println("someMethod $parameter")
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And .editorconfig
> > ```
> > [*.{kt,kts}]
> > ktlint_code_style = android_studio
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Producing
> > ```
> > /sample/Sample.kt:10:5: Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)
> > /sample/Sample.kt:11:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
> > /sample/Sample.kt:12:5: Single whitespace expected before parameter (standard:function-signature)
> > /sample/Sample.kt:12:35: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
> > /sample/Sample.kt:18:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
> > /sample/Sample.kt:18:23: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
> > ```
> 
> How can you be sure that your issue is identical to OP? For your problem, see [#2259 (comment)](https://github.com/pinterest/ktlint/issues/2259#issuecomment-1732333419) which will be fixed in `1.0.1` release.

Well, if you read the output carefully, you can see that the issues are not only related to Composable but are related to the standalone method as well.


Not sure why you renamed the issue because it is related to the original one. But I can open the same issue for you again if you wish so

**Comment by bjorn-jurgenn** [comment]
> > There is code sample
> > ```
> > import androidx.compose.foundation.layout.padding
> > import androidx.compose.material.Text
> > import androidx.compose.runtime.Composable
> > import androidx.compose.ui.Modifier
> > import androidx.compose.ui.unit.dp
> > 
> > @Composable
> > fun SampleComposable(
> >     parameter: String,
> >     modifier: Modifier = Modifier,
> > ) {
> >     Text(text = "Hello $parameter", modifier = Modifier.padding(24.dp))
> > }
> > 
> > fun someMethod(
> >     parameter: String,
> > ) {
> >     println("someMethod $parameter")
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And .editorconfig
> > ```
> > [*.{kt,kts}]
> > ktlint_code_style = android_studio
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Producing
> > ```
> > /sample/Sample.kt:10:5: Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)
> > /sample/Sample.kt:11:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
> > /sample/Sample.kt:12:5: Single whitespace expected before parameter (standard:function-signature)
> > /sample/Sample.kt:12:35: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
> > /sample/Sample.kt:18:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
> > /sample/Sample.kt:18:23: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
> > ```
> 
> How can you be sure that your issue is identical to OP? For your problem, see [#2259 (comment)](https://github.com/pinterest/ktlint/issues/2259#issuecomment-1732333419) which will be fixed in `1.0.1` release.

Please check the attached snippet carefully. It is not related to mentioned issue. Composable annotation is just one part of the problem. The false positives are logged for standalone method as well.

No need to close the issue when it’s not resolved.

**Comment by paul-dingemans** [comment]
> Please check the attached snippet carefully. It is not related to mentioned issue. Composable annotation is just one part of the problem. The false positives are logged for standalone method as well.
> 
> No need to close the issue when it’s not resolved.


OP's problem is:
> Ktlint produces an error about invalid ordering of imports, even though they are already lexographically organized without spaces.

I see no violations regarding imports in:
```
/sample/Sample.kt:10:5: Function name should start with a lowercase letter (except factory methods) and use camel case (standard:function-naming)
/sample/Sample.kt:11:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
/sample/Sample.kt:12:5: Single whitespace expected before parameter (standard:function-signature)
/sample/Sample.kt:12:35: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
/sample/Sample.kt:18:5: No whitespace expected between opening parenthesis and first parameter name (standard:function-signature)
/sample/Sample.kt:18:23: No whitespace expected between last parameter and closing parenthesis (standard:function-signature)
```

**Comment by pdb-stripe** [comment]
I will try to reproduce this when I have a chance. The error I specifically ran into was with the imports out of order. For now I am using 0.50, since ktlint-gradle does not work with ktlint 1.0.0

**Comment by bjorn-jurgenn** [comment]
> I will try to reproduce this when I have a chance. The error I specifically ran into was with the imports out of order. For now I am using 0.50, since ktlint-gradle does not work with ktlint 1.0.0

Are you sure that ktlint-gradle doesn’t work with ktlint 1.0.0?

**Comment by pdb-stripe** [comment]
@bjorn-jurgenn it doesn't work for me (See https://github.com/JLLeitschuh/ktlint-gradle/issues/714). It is possible this has to do with something in my own project's configuration, so it may work on other projects.

**Comment by kenyee** [comment]
FWIW, I found an interesting difference between the android_studio and intellij_idea formatting.
Intellij_idea forces you to put your java.* and javax.* imports at the bottom while the android_studio one doesn't and this different from the latest AS (Iguana Canary 9) and ktlint (1.0.1).
I also ended up using intellij_idea in .editorConfig to avoid ktlint CLI and AS fighting on import ordering.

**Comment by paul-dingemans** [comment]
> FWIW, I found an interesting difference between the android_studio and intellij_idea formatting. Intellij_idea forces you to put your java.* and javax.* imports at the bottom while the android_studio one doesn't and this different from the latest AS (Iguana Canary 9) and ktlint (1.0.1). I also ended up using intellij_idea in .editorConfig to avoid ktlint CLI and AS fighting on import ordering.

See [Ktlint documentation for configuring import layouts](https://pinterest.github.io/ktlint/1.0.1/rules/configuration-ktlint/#import-layouts).

**Comment by kenyee** [comment]
Thanks Paul.  Very interesting that even though the Android Studio style guide says formatting of imports is alphabetical:
https://developer.android.com/kotlin/style-guide#structure
it formats it the other way...or probably the ktlint IDE plugin (0.13) is doing this since 0.14 with ktlint 1.0.1 support isn't out yet.

**Comment by paul-dingemans** [comment]
The plugin does not affect import ordering. So if import ordering of ktlint is not in sync with Android Studio, then it should be fixed in Ktlint itself.

**Comment by kenyee** [comment]
hmm..ok.  I'll write up an issue w/ a small project sample to track this android_studio formatting discrepancy.

**Comment by pdb-stripe** [comment]
Sorry that it took so long to validate this with ktlint cli. I am attempting to update to the latest ktlint version (1.0.1) and the latest ktlint gradle plugin. When updating my editorconfig to include `ktlint_code_style = android_studio`, then running the ktlint cli tool, I still get the errors with imports. 

However - when looking very closely at these imports, it appears that this validation is accurate, however it does not align with default behavior of Android Studio / IntelliJ, which add `java.*` and `kotlin.*` imports at the end. This is related to #527
